### DISCUSSION TITLE
Add an option to return strings as pandas' string dtypes instead of objects in fetchdf()

### DISCUSSION THREAD

**Discussion by jiri-kulik** [duckdb/duckdb#2039] [Ideas]
Hi,
just discovered this amazing project. While testing the python api I noticed that fetchdf() always returns strings as object dtypes, even if a query is from a data frame with string dtypes. It would be useful to have a parameter in fetchdf() that would switch between returning strings as objects and string dtypes with a default being object.

**Comment by Alex-Monahan** [comment]
There is also the new and experimental Apache Arrow backed string type since Pandas 1.2! Maybe there could be an "allowed extension types" list parameter?
https://pandas.pydata.org/docs/reference/api/pandas.arrays.ArrowStringArray.html
